src/spatial/reaction_rates.jl

```diff
@@ -48,16 +54,21 @@ function total_site_rx_rate(rx_rates::RxRates, site)
 end
 
 """
-    update_rx_rates!(rx_rates, rxs, u, site)
+    update_rx_rates!(rx_rates, rxs, integrator, site)
 
 update rates of all reactions in rxs at site
 """
 function update_rx_rates!(rx_rates::RxRates, rxs, u::AbstractMatrix, integrator,
     site)
     ma_jumps = rx_rates.ma_jumps
     @inbounds for rx in rxs
-        rate = eval_massaction_rate(u, rx, ma_jumps, site)
-        set_rx_rate_at_site!(rx_rates, site, rx, rate)
+        if is_massaction(rx_rates, rx)
+            rate = eval_massaction_rate(u, rx, ma_jumps, site)
+            set_rx_rate_at_site!(rx_rates, site, rx, rate)
+        else
+            cr_jump = rx_rates.cr_jumps[rx - get_num_majumps(ma_jumps)]
+            set_rx_rate_at_site!(rx_rates, site, rx, cr_jump.rate(u, integrator.p, integrator.t, site))
```

Comment:
This isn't type stable is it? That is why we split the ConstantRateJump rates and affects in the non-spatial solvers and wrap them inside `FunctionWrapper`s.